code/game/objects/items/explosives/grenades/marines.dm

```diff
@@ -235,4 +235,159 @@ proc/flame_radius(radius = 1, turf/T, burn_intensity = 25, burn_duration = 25, b
 	name = "\improper Type 8 WP grenade"
 	desc = "A deadly gas grenade found within the ranks of the UPP. Designed to spill white phosporus on the target. It explodes 2 seconds after the pin has been pulled."
 	icon_state = "grenade_upp_wp"
-	item_state = "grenade_upp_wp"
\ No newline at end of file
+	item_state = "grenade_upp_wp"
+
+/obj/item/explosive/grenade/impact
+	name = "\improper M40 IMDP grenade"
+	desc = "A high explosive contact detonation munition utilizing the standard DP canister chassis. Has a focused blast specialized for door breaching and combating emplacements and light armoured vehicles. Capable of being loaded in the M92 Launcher, or thrown by hand."
+	icon_state = "grenade_impact"
+	item_state = "grenade_impact"
+	hud_state = "grenade_frag"
+	det_time = 40
+	dangerous = TRUE
+	underslug_launchable = TRUE
+
+/obj/item/explosive/grenade/impact/prime()
+	spawn(0)
+		explosion(loc, -1, 1, 0, 2)
+		qdel(src)
+	return
+
+/obj/item/explosive/grenade/impact/flamer_fire_act()
+	var/turf/T = loc
+	qdel(src)
+	explosion(T, -1, 1, 0, 2)
+
+/obj/item/explosive/grenade/impact/throw_impact(atom/hit_atom, speed)
+	. = ..()
+	if(active)
+		explosion(loc, -1, 1, 0, 2)
+		qdel(src)
+
+
+/obj/item/explosive/grenade/flare
+	name = "\improper M40 FLDP grenade"
+	desc = "A TGMC standard issue flare utilizing the standard DP canister chassis. Capable of being loaded in the M92 Launcher, or thrown by hand."
+	icon_state = "flare_grenade"
+	det_time = 0
+	throwforce = 1
+	dangerous = 0
+	underslug_launchable = TRUE
+	w_class = 2
+	hud_state = "grenade_frag"
+	var/fuel = 0
+
+/obj/item/explosive/grenade/flare/New()
+	fuel = rand(800, 1000) // Sorry for changing this so much but I keep under-estimating how long X number of ticks last in seconds.
```

Comment:
this is why ss13 needs a deltaseconds variable @mrstonedone

Reply:
no it doesn't.

Reply:
yes it does because then we can just do -1 * deltatime and get a reliable -1 per second trickle. unless world.time is calculated even when the game is frozen because thatd be dumb.

Reply:
The mc will call process every 2 seconds, no less, no more. If that occasionally happens 50 to 100 milliseconds late, you won't notice.

Reply:
@Qustinnus @MrStonedOne to be clear, that comment was copy pastaed; not my words